Makefile

```diff
@@ -186,14 +189,43 @@ docclean:
 
 RAWVER=$(shell $(PYTHON) tools/getnodeversion.py)
 VERSION=v$(RAWVER)
+ifdef NIGHTLY
+# For nightly builds, you must set NIGHTLY to "nightly", "next-nightly" or "rc".
+# For the nightly and next-nightly case, you need to set DATESTRING and COMMIT
+# in order to properly name the build.
+# For the rc case you need to set RC to an appropriate RC number
+ifeq ($(NIGHTLY),rc)
+ifndef RC
+$(error RC is not set)
+endif
+TAG=rc$(RC)
+else
+ifndef DATESTRING
+$(error DATESTRING is not set)
+endif
+ifndef COMMIT
+$(error COMMIT is not set)
+endif
+ifneq ($(NIGHTLY),nightly)
+ifneq ($(NIGHTLY),next-nightly)
+$(error NIGHTLY is not nightly or next-nightly)
+endif
+endif
+TAG=$(NIGHTLY)$(DATESTRING)$(COMMIT)
+endif
+FULLVERSION=$(VERSION)-$(TAG)
+DISTTYPE=$(NIGHTLY)
+else
 FULLVERSION=$(VERSION)
+DISTTYPE=release
+endif
 RELEASE=$(shell sed -ne 's/\#define NODE_VERSION_IS_RELEASE \([01]\)/\1/p' src/node_version.h)
 PLATFORM=$(shell uname | tr '[:upper:]' '[:lower:]')
 NPMVERSION=v$(shell cat deps/npm/package.json | grep '"version"' | sed 's/^[^:]*: "\([^"]*\)",.*/\1/')
 ifeq ($(findstring x86_64,$(shell uname -m)),x86_64)
 DESTCPU ?= x64
 else
-DESTCPU ?= ia32
+DESTCPU ?= x86
 endif
 ifeq ($(DESTCPU),x64)
 ARCH=x64
```

Comment:
In your example, you invoke make passing both DESTCPU=$DESTCPU and ARCH=$ARCH. Do we need to pass both or can one reliably be derived from the other?